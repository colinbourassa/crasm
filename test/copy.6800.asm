;;; Author: Leon Bottou
;;; Public Domain.

	cpu 6800
	
	* = $8000
	
	begin  = $40
	dest   = $42
	len    = $44
	
	
	ldx  #$4000
	stx  begin
	ldx  #$1430
	stx  len
	ldx  #$6000
	stx  dest
	jsr  copy
	wai
	
	code

	; copy LEN bytes from BEGIN to DEST
	
copy	ldx  begin
	sts  begin
	txs
	ldx  dest
	
	ldab len+1
	ldaa len
	addb dest+1
	adca dest
	stab dest+1
	staa dest
	
.1	cpx dest
	beq .2
	pula
	staa 0,x
	inx
	bra .1

.2	tsx
	lds begin
	stx begin
	clr len
	clr len+1
	rts
	
	
	code
