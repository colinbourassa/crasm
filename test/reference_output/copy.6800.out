S1138000CE4000DF40CE1430DF44CE6000DF42BDFE
S113801080133EDE409F4035DE42D6459644DB4326
S11380209942D74397429C42270632A7000820F67C
S10F8030309E40DF407F00447F00453953
S9030000FC
